### PR TITLE
refactor(authz): ensure roles/service-accounts follow the guidelines

### PR DESCRIPTION
**Pull request by H4ad**
## Pull Request

---

After guidelines are defined, we found few violations in the Roles page and Service Accounts page, since they are our standard/how to check permission, they should follow the standard strictly.

| Before | After |
|--------|--------|
| image | image |
| image | image |
| image | image |
| image | image |
| image | image | 
| image | image | 
| image | image | 
| image | image |
| image | image | 
| image | image | 

Closes https://github.com/SigNoz/platform-pod/issues/2833

---

### ✅ Change Type
_Select all that apply_

- [ ] ✨ Feature
- [ ] 🐛 Bug fix
- [x] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

- Tests added/updated: Yes
- Manual verification: Yes
- Edge cases covered: Yes

---

- Blast radius: Roles / Service Account
- Potential regressions: Forgot to migrate something or migrate something incorrectly.
- Rollback plan: Find and fix the issue

---

| Field | Value |
|------|-------|
| Deployment Type | Cloud / OSS / Enterprise |
| Change Type | Maintenance |
| Description | We updated the Roles and Service Account pages to ensure they follow our AuthZ Guidelines |

---

### 📋 Checklist
- [x] Tests added or explicitly not required
- [ ] Manually tested
- [ ] Breaking changes documented
- [ ] Backward compatibility considered

### CODE REVIEW

File: frontend/src/container/RolesSettings/ViewRolePage/ViewRolePage.tsx [hunk 1]
```diff
@@ -175,43 +268,95 @@ function ViewRolePageContent(): JSX.Element {
...
+				Delete
+			</AuthZButton>
 		);
-	}
+	};
 
-	if (!role) {
-		return <></>;
-	}
+	const renderUpdateButton = (): JSX.Element => {
```
Comment (AshwinBhatkal) [inline frontend/src/container/RolesSettings/ViewRolePage/ViewRolePage.tsx]: preferably different component. Seems like update and delete can also be 1 component

Comment (H4ad) [reply frontend/src/container/RolesSettings/ViewRolePage/ViewRolePage.tsx]: To join these two, I will need to create something like `renderActionButton = (variant, color, testId, className, onClick, permission) =>`

Not sure if worth the complexity when this is simple enough

Comment (H4ad) [reply frontend/src/container/RolesSettings/ViewRolePage/ViewRolePage.tsx]: Moved to header component

**Review by AshwinBhatkal — DISMISSED** [review]

**Review by AshwinBhatkal — APPROVED** [review]
